Dockerfile

```diff
@@ -19,17 +19,19 @@ ARG SPARK_IMAGE=gcr.io/spark-operator/spark:v2.4.0
 FROM golang:1.12.5-alpine as builder
 ARG DEP_VERSION="0.5.3"
 RUN apk add --no-cache bash git
-ADD https://github.com/golang/dep/releases/download/v${DEP_VERSION}/dep-linux-amd64 /usr/bin/dep
-RUN chmod +x /usr/bin/dep
 
-WORKDIR ${GOPATH}/src/github.com/GoogleCloudPlatform/spark-on-k8s-operator
-COPY Gopkg.toml Gopkg.lock ./
-RUN dep ensure -vendor-only
+WORKDIR /tmp/github.com/GoogleCloudPlatform/spark-on-k8s-operator
+COPY go.mod go.sum ./
+RUN go mod download
+
+FROM builder as operator_builder
+WORKDIR /tmp/github.com/GoogleCloudPlatform/spark-on-k8s-operator
 COPY . ./
+ENV GO111MODULE=on
```

Comment:
We need this because in the docker image we inherit from we have GOPATH and all deps wil be stored under $GOPATH/pgk/mod by default. From there we need to pickup the generator. Even if we run `go mod vendor` the vendor dir will not bring that dep in, because  its implicit.